permission/basic/backend.go

```diff
@@ -0,0 +1,368 @@
+				if err != nil {
+					log.Error("error updating permissioned-nodes.json", "err", err)
+				}
+				types.NodeInfoMap.UpsertNode(evtNodeApproved.OrgId, evtNodeApproved.EnodeId, types.NodeApproved)
+
+			case evtNodeProposed := <-chNodeProposed:
+				types.NodeInfoMap.UpsertNode(evtNodeProposed.OrgId, evtNodeProposed.EnodeId, types.NodePendingApproval)
+
+			case evtNodeDeactivated := <-chNodeDeactivated:
+				err := ptype.UpdatePermissionedNodes(b.Ib.Node(), b.Ib.DataDir(), evtNodeDeactivated.EnodeId, ptype.NodeDelete, b.Ib.IsRaft())
+				if err != nil {
+					log.Error("error updating permissioned-nodes.json", "err", err)
+				}
+				types.NodeInfoMap.UpsertNode(evtNodeDeactivated.OrgId, evtNodeDeactivated.EnodeId, types.NodeDeactivated)
+
+			case evtNodeActivated := <-chNodeActivated:
+				err := ptype.UpdatePermissionedNodes(b.Ib.Node(), b.Ib.DataDir(), evtNodeActivated.EnodeId, ptype.NodeAdd, b.Ib.IsRaft())
+				if err != nil {
+					log.Error("error updating permissioned-nodes.json", "err", err)
+				}
+				types.NodeInfoMap.UpsertNode(evtNodeActivated.OrgId, evtNodeActivated.EnodeId, types.NodeApproved)
+
+			case evtNodeBlacklisted := <-chNodeBlacklisted:
+				types.NodeInfoMap.UpsertNode(evtNodeBlacklisted.OrgId, evtNodeBlacklisted.EnodeId, types.NodeBlackListed)
+				err := ptype.UpdateDisallowedNodes(b.Ib.DataDir(), evtNodeBlacklisted.EnodeId, ptype.NodeAdd)
+				log.Error("error updating disallowed-nodes.json", "err", err)
+				err = ptype.UpdatePermissionedNodes(b.Ib.Node(), b.Ib.DataDir(), evtNodeBlacklisted.EnodeId, ptype.NodeDelete, b.Ib.IsRaft())
+				if err != nil {
+					log.Error("error updating permissioned-nodes.json", "err", err)
+				}
+
+			case evtNodeRecoveryInit := <-chNodeRecoveryInit:
+				types.NodeInfoMap.UpsertNode(evtNodeRecoveryInit.OrgId, evtNodeRecoveryInit.EnodeId, types.NodeRecoveryInitiated)
+
+			case evtNodeRecoveryDone := <-chNodeRecoveryDone:
+				types.NodeInfoMap.UpsertNode(evtNodeRecoveryDone.OrgId, evtNodeRecoveryDone.EnodeId, types.NodeApproved)
+				err := ptype.UpdateDisallowedNodes(b.Ib.DataDir(), evtNodeRecoveryDone.EnodeId, ptype.NodeDelete)
+				log.Error("error updating disallowed-nodes.json", "err", err)
+				err = ptype.UpdatePermissionedNodes(b.Ib.Node(), b.Ib.DataDir(), evtNodeRecoveryDone.EnodeId, ptype.NodeAdd, b.Ib.IsRaft())
+				if err != nil {
+					log.Error("error updating permissioned-nodes.json", "err", err)
+				}
+
+			case <-stopChan:
+				log.Info("quit Node Contr watch")
+				return
+			}
+		}
+	}()
+	return nil
+}
+
+func (b *Backend) MonitorNetworkBootUp() error {
+	netWorkBootCh := make(chan *bb.PermImplPermissionsInitialized, 1)
+
+	opts := &bind.WatchOpts{}
+	var blockNumber uint64 = 1
+	opts.Start = &blockNumber
+
+	if _, err := b.Contr.PermImpl.PermImplFilterer.WatchPermissionsInitialized(opts, netWorkBootCh); err != nil {
+		return fmt.Errorf("failed WatchNodeApproved: %v", err)
```

Comment:
Mistake in error message - should be "WatchPermissionsInitialized"

Reply:
done